src/plugins/mer/mersettings.cpp

```diff
@@ -123,6 +166,59 @@ QMap<QString, MerEmulatorDeviceModel> MerSettings::deviceModelsRead(const Utils:
     return result;
 }
 
+bool MerSettings::isDeviceModelStored(const MerEmulatorDeviceModel &model)
+{
+    Q_ASSERT(s_instance);
+
+    if (!s_instance->m_deviceModels.contains(model.name()))
+        return false;
+
+    return s_instance->m_deviceModels.value(model.name()) == model;
+}
+
+void MerSettings::setDeviceModels(const EmulatorDeviceModels &deviceModels)
+{
+    Q_ASSERT(s_instance);
+
+    if (s_instance->m_deviceModels == deviceModels)
+        return;
+
+    //ToDo: use std::transform_reduce (C++17)
+    const QMap<QString, MerEmulatorDeviceModel> modelsChanged {
+        Utils::filtered(s_instance->m_deviceModels.toStdMap(),
+                        [&deviceModels](const Map::value_type &pair) {
+            const QString name = pair.first;
```

Comment:
If your motivation here was code effectiveness, then I believe using `pair.first` inline would be more effective (it is passed by reference). And maybe it does not matter at all what you do, because compilers are likely to optimize out the extra copy you introduced. Anyway in this kind of code readability counts more than speed of execution :) So if your motivation was readability, then maybe renaming `pair` as `nameValuePair` would do better job.

Reply:
There should be lazy copy - fast empty constructor with one pointer assignment, I presume. And I agree that compiler should optimize this variable. In this particularly case my intention is less typing. I renamed `pair` to `nameValuePair`.

Reply:
one pointer assignment + incrementing a reference counter, similarly the destructor would decrement it :)